src/main/java/org/folio/service/orders/CompositePoLineValidationService.java

```diff
@@ -76,6 +76,58 @@ private List<Error> validatePoLineFormats(CompositePoLine compPOL) {
     return Collections.emptyList();
   }
 
+  public List<Error> validatePoLineMaterial(CompositePoLine compPOL) {
+    CompositePoLine.OrderFormat orderFormat = compPOL.getOrderFormat();
+
+    return switch (orderFormat) {
+      case P_E_MIX -> checkPEMix(compPOL);
+      case ELECTRONIC_RESOURCE -> checkElectronicResource(compPOL);
+      case PHYSICAL_RESOURCE -> checkPhysicalResource(compPOL);
+      case OTHER -> checkOtherResource(compPOL);
+      default -> Collections.emptyList();
+    };
+  }
+
+  private List<Error> checkPEMix(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getPhysical() == null) || (compPOL.getEresource() == null) || (getElectronicCostQuantity(compPOL) == 0) || (getPhysicalCostQuantity(compPOL) == 0)) {
+      errors.add(ErrorCodes.INVALID_PEMIX_POL);
+    }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkElectronicResource(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getPhysical() != null) && (getPhysicalCostQuantity(compPOL) != 0)) {
+        errors.add(ErrorCodes.INVALID_ELECTRONIC_POL);
+      }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkPhysicalResource(CompositePoLine compPOL ){
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getEresource() != null) && (getElectronicCostQuantity(compPOL) != 0)) {
+        errors.add(ErrorCodes.INVALID_PHYSICAL_POL);
+      }
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  private List<Error> checkOtherResource(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    if ((compPOL.getEresource() != null) && (getElectronicCostQuantity(compPOL) != 0)) {
+        errors.add(ErrorCodes.INVALID_OTHER_POL);
+      }
```

Comment:
Two conditions are the same, please check that again, just in case it seems strange to me

Reply:
thank you, Azizbek, :) 
yes, both the other and physical ordering formats should have physical resource only, so I both of them have the same condition.